client/src/StudentHomePage/StudentHomePage.js

```diff
@@ -8,34 +16,93 @@ class StudentHomePage extends React.Component{
+                {
+                    rubric_id: 100,
+                    assigned_date: "2020-04-21T07:00:00.000Z ",
+                    due_date: "2020-04-28T07:00:00.000Z",
+                    final_due_date: "2020-05-01T07:00:00.000Z ",
+                    section_id: 1 
+                },
+                {
+                    rubric_id: 200,
+                    assigned_date: "2020-04-21T07:00:00.000Z ",
+                    due_date: "2020-04-28T07:00:00.000Z",
+                    final_due_date: "2020-05-01T07:00:00.000Z ",
+                    section_id: 1 
+                }
+            ],
+            isHovered: false 
         }
+        
+        this.toggleHover = this.toggleHover.bind(this);
+    }
+
+    toggleHover(){
+        this.setState(prevState => ({isHovered: !prevState.isHovered}));
+    }
+
+
+    componentDidMount(){
+    	axios({
+            method: 'get',
+            url:'https://rwlautograder.herokuapp.com/api/stud/registered/class/info'
+        }).then(res => {
+    		const studentInfo = res.data;
+            this.setState({ studentInfo });
+            console.log(this.state.studentInfo)
+        }).catch((error) =>{
+            if(error.response){
+              console.log(error.response.data);
+            } else if (error.request){
+                console.log(error.request); 
+            }else {
+                console.log(error.message);
+            }
+        })
+
+        axios({
+            method: 'get',
+            url: 'https://rwlautograder.herokuapp.com/api/stud/class/' + this.state.studentInfo[0].class_id + '/assignment/dates'
+        }).then(res => {
+            this.setState({ assignments : res.data});
+            console.log(this.state.assignments)
+        }).catch((error) =>{
+            if(error.response){
+              console.log(error.response.data);
+            } else if (error.request){
+                console.log(error.request); 
+            }else {
+                console.log(error.message);
+            }
+        })
     }
 
     tableBody(){
         return (
-            this.table = this.state.info.map((data) => 
-                <tr onClick={ function() {window.alert(data.assignment)}}>
-                    <td><b>{data.todo}</b> {data.assignment}</td>
-                    <td>{data.class_name}</td>
-                    <td>{data.due_date}</td>
+            this.table = this.state.assignments.map((data, i) => 
+                <tr>
+                      <td><div onMouseEnter={this.toggleHover} onMouseLeave={this.toggleHover}><b>Homework {i+1}</b><br/>{this.state.isHovered ? "" : <i>{data.assigned_date}</i> }</div></td>
+                      <td><Link to={{pathname: '/student/submit/' + (i+1) , state: {todo: (i+1), rubricID: data.rubric_id}}}>{data.due_date}</Link></td>
+                      <td><Link to={{pathname: '/student/grade/' + (i+1), state: {todo: (i+1), rubricID: data.rubric_id}}}>{data.final_due_date}</Link></td>
```

Comment:
For Dates we can format them so they look human readable. 
Resources:
* https://www.w3schools.com/js/js_date_formats.asp
* https://www.npmjs.com/package/dateformat
You probably want to do something like 
```
dateFormat(new Date(data.*date), "dddd, mmmm dS, yyyy, h:MM:ss TT");
```